Source/Csla.Shared/DataPortal.cs

```diff
@@ -873,5 +873,35 @@ public static void ResetProxyType()
     [Obsolete("Proxies no longer cached")]
     public static void ReleaseProxy()
     { }
+
+    internal static object GetCriteriaFromArray(params object[] criteria)
+    {
+      if (criteria == null)
+        return null;
+      else if (criteria.GetLength(0) == 0)
```

Comment:
You call `criteria.GetLength(0)` potentially 3 times. Small thing, but could you grab that in a local variable and then just use that?

Reply:
Seems reasonable